File: frontend/src/microservices/attraction/objects/CreateAttractionTicketTypePlannerRequest.ts
export type CreateAttractionTicketTypePlannerRequest = {
  managerId: string
  attractionId: string
  ticketTypeName: string
  description: string
  unitPrice: string
  currency: string
  availableFromDate: string
  availableToDate: string
  totalQuantity: number
  validWeekdays: string[]
}

export const createAttractionTicketTypePlannerRequestFromJson = (json: string): CreateAttractionTicketTypePlannerRequest =>
  JSON.parse(json) as CreateAttractionTicketTypePlannerRequest

export const createAttractionTicketTypePlannerRequestToJson = (value: CreateAttractionTicketTypePlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/CreateAttractionTicketTypePlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateAttractionTicketTypePlannerRequest(
    managerId: String,
    attractionId: String,
    ticketTypeName: String,
    description: String,
    unitPrice: String,
    currency: String,
    availableFromDate: String,
    availableToDate: String,
    totalQuantity: Int,
    validWeekdays: List[String]
)
object CreateAttractionTicketTypePlannerRequest:
  given sourceEncoder: Encoder[CreateAttractionTicketTypePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateAttractionTicketTypePlannerRequest] = deriveDecoder
